### DISCUSSION TITLE
Harmony Template deprecation

### DISCUSSION THREAD

**Discussion by wish79** [primefaces/community#2453] [PrimeFaces Templates]
Dear Primetek,
We have been using the Primefaces Harmony Template since 2018. Recently, we noticed that the [Harmony Template has been deprecated](https://github.com/orgs/primefaces/discussions/1803) along with some other templates. We have the following questions regarding this matter:
1.	Will you continue to support the Harmony Template with future Primefaces version updates?
2.	If support for the Harmony Template will not be provided, how can we ensure that the template will remain compatible with new Primefaces versions as we upgrade?
3.	Should we consider switching to a different template? If so: a. Which template would be the best alternative that closely resembles Harmony? b. Will you provide us with the new template at no additional cost, given that we have already paid for the Harmony Template?
We are disappointed by the deprecation of the Harmony Template, as this will require significant effort to either upgrade or maintain the current version.

Thank you for your assistance.

Thanks
Mohd

**Comment by melloware** [comment]
@wish79 i don't work for PrimeTek but here are some answers I have seen and given before.

1. Don't expect any new releases in your PrimeTek store account for Harmony.   
2. They provide instructions for upgrading the theme yourself: https://github.com/primefaces/primefaces/wiki/Migration-Guide-for-Templates  I have upgraded Atlantis, Ultima, Barcelona, California etc to PF 14.   If you need any help you can shoot me a DM on the Discord channel.
3. This answer I would wait and see on.  It looks like right now they have a handful of templates they still support like Diamond etc but whether you should migrate or not is up to you.  I personally wouldn't I would just update your current Harmony theme...

**Comment by tandraschko** [comment]
PrimeTek decided to only maintain the following templates for now:
- "apollo"
- "atlantis"
- "avalon"
- "diamond"
- "freya"
- "poseidon"
- "ultima"
- "verona"
